### PR TITLE
[B5] Bootstrap 5 styling updates for select2

### PR DESCRIPTION
**Pull request by biyeun**
## Technical Summary
Covered here are updates to `select2` `scss` and the `hqwebapp-select2` widget that are only relevant to `select2` instances used on bootstrap 5 pages. Currently the only instances of these are in the WIP styleguide.

### Safety story
Safe change due to limited scope and that it's not being used in any current production workflows

### Automated test coverage
Yes, diffs are covered.

### QA Plan
Not needed due to limited scope.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss [hunk 1]
```diff
@@ -73,3 +73,86 @@
 .select2-selection__placeholder {
   color: $gray-base !important;
 }
+
+.select2-container--default .select2-selection--single,
+.select2-container--default .select2-selection--multiple {
+  border-color: $border-color !important;
+  height: 32px !important;
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss:80]: Is this to align select2 height with the height of text inputs?

Comment (biyeun) [reply corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss:80]: yes

Comment (biyeun) [reply corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss:80]: the text inputs in bootstrap 5 use a y-padding value, but select2 goes by a specific height

Comment (orangejenny) [reply corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss:80]: awesome

**Comment by biyeun** [comment]
bump on review :)

**Review by orangejenny — APPROVED** [review]

**Pull request merged**